crates/forge/src/collecting.rs

```diff
@@ -5,13 +5,13 @@ use assert_fs::TempDir;
 use cairo_lang_sierra::program::Program;
 use camino::{Utf8Path, Utf8PathBuf};
 use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
-use test_collector::{collect_tests, LinkedLibrary, TestCase};
+use test_collector::{collect_tests, ForkConfig, LinkedLibrary, RawForkConfig, TestCase};
 use walkdir::WalkDir;
 
 #[derive(Debug, Clone)]
-pub(crate) struct CompiledTestCrate {
+pub(crate) struct CompiledTestCrate<T: ForkConfig> {
     pub sierra_program: Program,
-    pub test_cases: Vec<TestCase>,
+    pub test_cases: Vec<TestCase<T>>,
     pub tests_location: CrateLocation,
 }
```

Comment:
This parametrization only introduces complexity without any direct benefit. As I wrote below, we should always parse fork configs and only return test cases with it parsed, no parametrization needed, especially since basically all code post test collection is coded to expect `TestCase<ValidatedForkConfig>>`.